packages/bsb-core/bsb/simulation/adapter.py

```diff
@@ -1,49 +1,73 @@
 import abc
-import itertools
-import types
+import os
+import sys
 import typing
 from contextlib import ExitStack
 from time import time
 
-import numpy as np
+from bsb import AttributeMissingError, SimulationResult, options, report
 
 from ..services.mpi import MPIService
-from .results import SimulationResult
 
 if typing.TYPE_CHECKING:
     from ..storage import PlacementSet
     from .cell import CellModel
     from .simulation import Simulation
 
 
-class AdapterProgress:
-    def __init__(self, duration):
-        self._duration = duration
+class BasicSimulationListener:
```

Comment:
"Basic" doesn't describe what it does

```suggestion
class FixedStepProgressController:
```